test/unit/lodash.spec.js

```diff
@@ -0,0 +1,522 @@
+            */
+        });
+
+        describe('min', function () {
+
+            var params;
+
+            beforeEach(function () {
+                params = datasets.min();
+            });
+
+            it('should find min value (-3) in user array', function () {
+                expect(params).to.have.length(1);
+                _(params[0]).forEach(function (value) {
+                    expect(typeof value).to.eql('number');
+                });
+                expect(_.min.apply(_, params)).to.eql(-3);
+            });
+        });
+
+
+        describe('minBy', function () {
+            
+            describe('minBy - Array', function () {
+
+                var params;
+
+                beforeEach(function () {
+                    params = datasets.minBy1();
+                });
+
+                it('should verify params', function () {
+                    expect(params).to.have.length(1);
+                    expect(params[0] instanceof Array).to.eql(true);
+                });
+
+                it('should return min value in Array', function () {
+                    expect(_.min.apply(_, params)).to.eql(-1);
+                });
+            });
+
+            /*
+            describe('minBy - Function', function () {
+                it('Function', function () {
+                });
+            });
+            //I have no idea how to do it. Probably minBy doesn't support function. Help!
+            */
+
+            describe('minBy - Object', function () {
+
+                var params;
+
+                beforeEach(function () {
+                    params = datasets.minBy3();
+                });
+                
+                it('should verify params', function () {
+                    expect(params).to.have.length(2);
+                    expect(params[0] instanceof Object).to.eql(true);
+                    expect(typeof params[1]).to.eql('string');
+                });
+
+                it('should return min value from Object', function () {
+                    expect(_.min.apply(_, params)).to.eql({ Name: 'Jack', Gold: 1500 });
+                });
+            });
+
+            describe('minBy - String', function () {
+
+                var params;
+
+                beforeEach(function () {
+                    params = datasets.minBy4();
+                });
+                
+                it('should verify params', function () {
+                    expect(params).to.have.length(1);
+                    expect(typeof params[0]).to.eql('string');
+                });
+                
+                it('should return min value from string', function () {
+                    expect(_.min.apply(_, params)).to.eql('e');
+                });
+            });
+        });
+
+        describe('multiply', function () {
+
+            var val1;
+            var val2;
+            var result1;
+            var params = datasets.multiply();
+
+            beforeEach(function () {
```

Comment:
With randoms it isn't that obvious but looking at the test cases it is the same as line 17.